[WIP] Switch to Microsoft.Azure.ServiceBus

0xced opened the pull request:
The [`Microsoft.Azure.ServiceBus`](https://www.nuget.org/packages/Microsoft.Azure.ServiceBus/) package conforms to .NET Standard and thus works fine on Linux and macOS (using Mono). The WindowsAzure.ServiceBus package can't run on Mono since it depends on kernel32.dll.

Required incidental changes:

* Upgrade target framework from v4.6 to v4.6.1
* Swith to async code
* Upgrade `Newtonsoft.Json` from version 9 to version 11
* Delete IsBodyConsumed and State for which I have found no equivalent
* Take a dependency on `Mossharbor.AzureWorkArounds.ServiceBus` since management is not (yet) supported with a service bus connection string, see https://github.com/Azure/azure-service-bus-dotnet/issues/65
* ~~Use a local version of `Mossharbor.AzureWorkArounds.ServiceBus` since `GetQueues()` and `GetTopics()` are not implemented in the published package https://github.com/Mossharbor/AzureWorkArounds.ServiceBus/pull/6~~

~~Note that this pull request is not ready to merge because of the local version of `Mossharbor.AzureWorkArounds.ServiceBus` which must be downloaded and compiled on its own.~~

0xced committed:
[WIP] Switch to Microsoft.Azure.ServiceBus

The [`Microsoft.Azure.ServiceBus`](https://www.nuget.org/packages/Microsoft.Azure.ServiceBus/) package conforms to .NET Standard and thus works fine on Linux and macOS (using Mono). The WindowsAzure.ServiceBus package can't run on Mono since it depends on kernel32.dll.

Required incidental changes:

* Upgrade target framework from v4.6 to v4.7.1
* Swith to async code
* Upgrade `Newtonsoft.Json` from version 9 to version 10
* Delete IsBodyConsumed and State for which I have found no equivalent
* Take a dependency on `Mossharbor.AzureWorkArounds.ServiceBus` since management is not (yet) supported with a service bus connection string, see https://github.com/Azure/azure-service-bus-dotnet/issues/65
* Use a local version of `Mossharbor.AzureWorkArounds.ServiceBus` since `GetQueues()` and `GetTopics()` are not implemented in the published package https://github.com/Mossharbor/AzureWorkArounds.ServiceBus/pull/6

0xced committed:
Remove Microsoft.WindowsAzure.ConfigurationManager

drewfreyling commented:
@0xced love the work you have done here. The downside of this is that it removes support of on premise Windows Service Bus. Having said that, WSB is [dead](https://stackoverflow.com/a/38769190/122880) so now should be the time we consider to drop support for it.

Mossharbor commented:
@0xced Mossharbor.AzureWorkArounds.ServiceBus GetQueues and GetTopics should be available in version *.15 through nuget, thanks to your pull request.

0xced committed:
Use Mossharbor.AzureWorkArounds.ServiceBus 1.0.15

0xced committed:
Downgrade from C# 6 to C# 5

Continuous integration on AppVeyor uses C# 5

0xced committed:
Delete unused string extensions (MakeDeadLetterPath)

Microsoft.Azure.ServiceBus now provides this feature with `EntityNameHelper.FormatDeadLetterPath()`

0xced committed:
Remove dead letter path when performing `RequeueOne`

This was omitted when refactoring to use Microsoft.Azure.ServiceBus

0xced committed:
Use Serilog.Sinks.Console instead of Serilog.Sinks.Literate

The features of Serilog.Sinks.Literate have now been merged into the default Serilog console sink.

The motivation behind this change is to be able to easily change the theme `WriteTo.Console(theme: ConsoleTheme.None)` so that the output in the JetBrains Rider console isn't white text on white background.

0xced committed:
Upgrade to Newtonsoft.Json version 11.0.2 in all projects

0xced committed:
Target .NET Framework version 4.6.1

The Microsoft.Azure.ServiceBus library targets .NET Standard 2.0 which is actually compatible with .NET Framework 4.6.1, see https://github.com/dotnet/standard/blob/release/2.0.0/docs/versions.md

0xced commented:
Like #93, passing continuous integration testing would require to upgrade AppVeyor image to a more recent version of Visual Studio.

0xced committed:
Upgrade to Mossharbor.AzureWorkArounds.ServiceBus 1.0.19

0xced committed:
Convert solution to the new csproj format

Although the [documentation (Additions to the csproj format for .NET Core)][1] seems to imply that this is a .NET Core feature, it works with .NET Framework too and is much easier to maintain.

With this commit, the OctoPack feature is gone because the new .csproj format is not yet supported by OctoPack. Here is the [suggested workaround][2]:
> We have this support on our radar, but don't have an ETA at this stage.
In the meantime, you could try running `dotnet publish` and then run `octo.exe pack` against the publish output folder.

[1]: https://docs.microsoft.com/en-us/dotnet/core/tools/csproj
[2]: https://github.com/OctopusDeploy/Issues/issues/3822#issuecomment-333706392

# Conflicts:
#	src/SbManager/SbManager.csproj
#	src/SbManager/packages.config

0xced committed:
Optimize getting a single queue/topic/subscription

0xced commented:
Note: dependency on `Mossharbor.AzureWorkArounds.ServiceBus` can be removed now that `Microsoft.Azure.ServiceBus` 3.1.0 supports management operations, see https://github.com/Azure/azure-service-bus-dotnet/pull/481

drewfreyling commented:
> You may as well tag/fork/whatever the last build that supports SBWS for anyone that cares and move on
> […](#)
> On Wed, May 30, 2018 at 9:45 AM, Drew Freyling ***@***.***> wrote: @0xced <https://github.com/0xced> love the work you have done here. The downside of this is that it removes support of on premise Windows Service Bus. Having said that, WSB is dead <https://stackoverflow.com/a/38769190/122880> so now should be the time we consider to drop support for it. — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub <[#91 (comment)](https://github.com/GlobalX/SbManager/pull/91#issuecomment-392983409)>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AA71rI4TXPDndxgVrysEtMx1G295kCOOks5t3d2FgaJpZM4USSA5> .
@lukeschafer done - https://github.com/GlobalX/SbManager/tree/WSB

drewfreyling commented:
Now that WSB is out of the way this PR can proceed.

0xced committed:
Switch to built-in ManagementClient

Drop Mossharbor.AzureWorkArounds.ServiceBus now that management operations are supported in the Microsoft.Azure.ServiceBus library.

0xced committed:
Delete the useless WindowsService class

Since WindowsService doesn't use the `HostControl` parameter, we can use IService directly.

0xced commented:
Could you please update the AppVeyor configuration to use VS 2017 instead of VS 2015?

0xced committed:
Use a relative path for the connection string file

0xced committed:
Load the connection string from a file if it exists

0xced committed:
Fix infinite loading when an error occurs while peeking messages

0xced committed:
Remove unused /api/v1/busmanager/deadletters endpoint

0xced committed:
Fix path unescaping

Because _ is a valid character in a queue, topic or subscription
